### PR TITLE
Adapt to Boost 1.70

### LINKED ISSUE #3851: Failure in CGAL_SetupBoost.cmake with boost 1.70.0
## Issue Details

On macOS (all versions), we are seeing the following issue with since the release of Boost 1.70.0. Any `cmake` invocation using `find_package(CGAL)` fails with:

```
-- The C compiler identification is AppleClang 10.0.1.10010046
-- The CXX compiler identification is AppleClang 10.0.1.10010046
-- Check for working C compiler: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang
-- Check for working C compiler: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++
-- Check for working CXX compiler: /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- __cplusplus is 199711
-- Found Boost 1.70.0 at /usr/local/lib/cmake/Boost-1.70.0
--   Requested configuration: QUIET REQUIRED COMPONENTS thread;system
-- Found boost_headers 1.70.0 at /usr/local/lib/cmake/boost_headers-1.70.0
-- Found boost_thread 1.70.0 at /usr/local/lib/cmake/boost_thread-1.70.0
-- No suitable boost_thread variant has been identified!
--   libboost_thread-mt.dylib (shared, BUILD_SHARED_LIBS not ON, set Boost_USE_STATIC_LIBS=OFF to override)
--   libboost_thread-mt.a (static, Boost_USE_STATIC_LIBS=OFF)
CMake Error at /usr/local/lib/cmake/Boost-1.70.0/BoostConfig.cmake:95 (find_package):
  Found package configuration file:

    /usr/local/lib/cmake/boost_thread-1.70.0/boost_thread-config.cmake

  but it set boost_thread_FOUND to FALSE so package "boost_thread" is
  considered to be NOT FOUND.  Reason given by package:

  No suitable build variant has been found.

Call Stack (most recent call first):
  /usr/local/lib/cmake/Boost-1.70.0/BoostConfig.cmake:124 (boost_find_dependency)
  /usr/local/Cellar/cmake/3.14.2_1/share/cmake/Modules/FindBoost.cmake:266 (find_package)
  /usr/local/lib/cmake/CGAL/CGAL_SetupBoost.cmake:48 (find_package)
  /usr/local/lib/cmake/CGAL/CGALConfig.cmake:87 (include)
  /usr/local/lib/cmake/CGAL/CGALConfig.cmake:142 (check_cgal_component)
  CMakeLists.txt:59 (find_package)
```

This was seen as part of [Homebrew testing for Boost 1.70.0](https://github.com/Homebrew/homebrew-core/pull/38890). We are using CGAL 4.14 and cmake 3.14.2 and the system compilers. The issue does not occur with Boost 1.69.0.

### PR DESCRIPTION
**Pull request by lrineau**
## Summary of Changes

Boost 1.70 now ships with CMake config files, and thus `find_package(Boost)` will work in `Config` mode, now.

This set of changes fixes various issues with that new setup.

## Release Management

* Affected package(s): Installation
* Issue(s) solved (if any): fix #3851 
* License and copyright ownership: maintenance by GF

### CODE REVIEW

**Comment by maxGimeno** [comment]
Travis still failing.

**Comment by lrineau** [comment]
Good. The last commit (https://github.com/CGAL/cgal/pull/3886/commits/02dd93fc27ed082053b562dcf0a6a9cf5dfc0ed6) fixed the last issue: this Travis job is not finished, but the `Classification` line passed: https://travis-ci.org/CGAL/cgal/jobs/526347824.

**Comment by maxGimeno** [comment]
I think there are still problems with boost [here](https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-29.shtml#Jet_fitting_3_Examples). But it is boost 1.71 from the master branch, i don't know if it is an error from us or from boost.

**Comment by maxGimeno** [comment]
Conflicts Polyhedron/demo/Polyhedron/Plugins/Mesh_3/CMakeLists.txt, Polyhedron/demo/Polyhedron/Plugins/Classification/CMakeLists.txt, Installation/cmake/modules/CGAL_SetupCGAL_CoreDependencies.cmake, 
and .travis/build_package.sh when trying to merge into integration

**Comment by lrineau** [comment]
I'll do the first merge in `integration`.

File: Polyhedron/demo/Polyhedron/Plugins/Mesh_3/CMakeLists.txt [hunk 1]
```diff
@@ -1,55 +1,51 @@
...
-      endif()
+remove_definitions(-DQT_STATICPLUGIN)
+
+qt5_wrap_cpp( VOLUME_MOC_OUTFILES ${CMAKE_CURRENT_SOURCE_DIR}/Volume_plane_thread.h )
+qt5_wrap_cpp( VOLUME_MOC_OUTFILES ${CMAKE_CURRENT_SOURCE_DIR}/Volume_plane_interface.h )
+qt5_wrap_ui( meshingUI_FILES  Meshing_dialog.ui Smoother_dialog.ui Local_optimizers_dialog.ui )
+polyhedron_demo_plugin(mesh_3_plugin Mesh_3_plugin
+  Mesh_3_plugin_cgal_code.cpp Meshing_thread.cpp split_polylines.cpp
+  ${meshingUI_FILES})
+target_link_libraries(mesh_3_plugin PUBLIC scene_polyhedron_item scene_polygon_soup_item scene_polylines_item scene_implicit_function_item scene_image_item
```
Comment (maxGimeno) [inline Polyhedron/demo/Polyhedron/Plugins/Mesh_3/CMakeLists.txt]: That cannot work in master, there is no more scene_polyhedron_item target

Comment (lrineau) [reply Polyhedron/demo/Polyhedron/Plugins/Mesh_3/CMakeLists.txt]: I was wrong when I did the merge. Sorry.

**Pull request merged**